### DISCUSSION TITLE
Desktop shortcut to open the latest IDE version always

### DISCUSSION THREAD

**Discussion by Greedquest** [twinbasic/twinbasic#1258] [Show and tell]
Make a shortcut and set the target to be the following:

```
%ComSpec% /v /r "for /L %L in (999 -1 1)do if exist "%Userprofile%\Downloads\twinBASIC_IDE_BETA_%~L\." (cd /d "%Userprofile%\Downloads\twinBASIC_IDE_BETA_%~L" && start "" /b "!cd!\twinBASIC.exe" && exit /b)"
```

Set it to [run minimised](https://user-images.githubusercontent.com/25348920/195874052-4158a8b9-79f1-4f28-bd8a-d8affe78b587.png) if you like

Now whenever you download a beta release and unzip it in your downloads folder, the shortcut will point to that version.

 > `%Userprofile%\Downloads\twinBASIC_IDE_BETA_123\twinBASIC.exe`

All credit to [Io-oI](https://superuser.com/users/969781/io-oi) on Superuser [[link](https://superuser.com/a/1747424/705597)] -  but it's a bit of a "gimme the code" question I asked and will probably be deleted soon. RIP

---

```
%ComSpec% /v /r "for /f delims^=* %i in ('dir /b /s /a:d /o-d "%Userprofile%\Downloads\twinBASIC_IDE_BETA_*"')do if exist "%~dpnxi\twinBASIC.exe" (cd /d "%~dpnxi" && start "" /b "!cd!\twinBASIC.exe" && exit /b)"
```
searches by creation date (most recent downloaded ide)

**Comment by WaynePhillipsEA** [comment]
Nice!

**Reply by wqweto** [reply]
Can we get [this](https://github.com/twinbasic/twinbasic/issues/1003) auto-registration baked in on startup iff the EXE started is a newer version than the one registered (or the registered executable is missing/unreadable).

I was recently scratching my head why some mildly recent language construct was not working with BETA 48 :-))

**Comment by mburns08109** [comment]
I always just copy the .zip contents over to a common folder and allow it to overwrite as needed. no need to wrestle with this, then.

**Reply by Greedquest** [reply]
Yeah that was the other suggestion in the linked superuser question, but it's marginally more effort and you lose track of which version you are using because it's not in the original named folder. I just thought this was a neat thing to share.
